src/argilla_server/schemas/v1/records.py

```diff
@@ -200,6 +200,21 @@ class RecordsUpdate(BaseModel):
     items: List[RecordUpdateWithId] = Field(..., min_items=RECORDS_UPDATE_MIN_ITEMS, max_items=RECORDS_UPDATE_MAX_ITEMS)
 
 
+class RecordUpsert(BaseModel):
+    fields: Optional[Dict[str, str]]
+    metadata: Optional[Dict[str, Any]]
+    external_id: Optional[str]
+    id: Optional[UUID]
+
+
+class RecordsBulkUpsert(BaseModel):
+    items: conlist(item_type=RecordUpsert, min_items=RECORDS_CREATE_MIN_ITEMS, max_items=RECORDS_CREATE_MAX_ITEMS)
```

Comment:
Can we instead use a `Field`?

I would also add `BULK` to the constants names so it's clear this is for bulk:

```suggestion
    items: List[RecordUpsert] = Field(
        ...,
        min_items=RECORDS_BULK_CREATE_MIN_ITEMS,
        max_items=RECORDS_BULK_CREATE_MAX_ITEMS,
    )
```